cosmic-core/server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -3337,7 +3337,7 @@ public boolean releaseSecondaryIpFromNic(final long ipAddressId) {
                 }
             }
             //check if the secondary ip associated with any static nat rule
-            final IPAddressVO publicIpVO = _ipAddressDao.findByVmIp(secondaryIp);
+            final IPAddressVO publicIpVO = _ipAddressDao.findByIpAndNetworkId(secIpVO.getNetworkId(), secondaryIp);
```

Comment:
@eliasgomes The network type `Network.GuestType.Isolated` is also used for VPC tier networks.